feature(swap): Steelswap aggregator

jorbuedo opened the pull request:
<!-- CURSOR_SUMMARY -->
> [!NOTE]
> Adds Steelswap aggregator (API, transformers, types) and integrates it into routing and UI, while refining Minswap transformers, protocol mappings, and types.
> 
> - **Swap Adapters**:
>   - **Steelswap**: New adapter with `api-maker`, `transformers`, `types`, and docs; handles tokens, orders, estimates, create, cancel; mainnet-only; custom error parsing.
>   - **Minswap**: Simplify requests (remove `blockedProtocols`/`exclude_protocols`), adjust fee handling (`dexFee` vs `aggregatorFee`, precise `totalFee`), and update protocol mapping for `splash`.
> - **Manager Integration**:
>   - Add Steelswap to `swap/manager.ts` with partner-based enabling; dynamic adapter set; cancel flow considers Steelswap; safer handling when adapters are absent.
> - **UI**:
>   - Estimate route shows aggregator (e.g., "via Steelswap") and multi-split ellipsis.
>   - Swap settings: routing preferences now driven by Remote Config partners; dynamic toggles and display names.
>   - Actions banner: safe access to `initialPair?.tokenOut`.
> - **Types & Protocols**:
>   - Add `Swap.Aggregator.Steelswap`.
>   - Update `Swap.Protocol.Splash_v1` to `"splash"`.
>   - Relax `YoroiConfig` optional fields; include `swap.partners.steelswap`.
> - **Docs**: Add `steelswap.api.md` with base URL, auth, supported DEXes.
> - **Versions**: Bump `@yoroi/swap` to `7.2.0` and `@yoroi/types` to `6.2.1`.
> - **Tests**: Add comprehensive Steelswap tests; update Minswap tests accordingly.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit d1c7e91fe4bde7ca80045dea51a46c654846fff9. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

jorbuedo committed:
add steelswap agregator

jorbuedo committed:
fix fee sum precission

jorbuedo committed:
fix no limit

jorbuedo committed:
update packages

jorbuedo committed:
Merge remote-tracking branch 'origin/develop' into steelswap

# Conflicts:
#	scripts/packages/swap/package.json

jorbuedo commented on mobile/packages/types/swap/protocol.ts:
```diff
@@ -16,7 +16,7 @@ export const SwapProtocol = Object.freeze({
   Wingriders_v2: 'wingriders-v2',
   Wingriders_stable: 'wingriders-stable',
-  Splash_v1: 'splash-v1',
+  Splash_v1: 'splash',
```
to double check
The idea is I don't want OUR type to include the v1 version internally because some aggregators just use the latest one and don't want to show in the UI this v1 or v4 stuff, except for very major ones like minswap v1 and v2, or wingriders v1 and v2, because those are very public and still both used. 
I believe this change doesn't really pose a problem as each aggregator maps it to what they use anyway

jorbuedo committed:
fix test

jorbuedo committed:
fix

jorbuedo committed:
fix(steelswap): remove incorrect timestamp multiplication

The Steelswap API returns timestamps already in milliseconds (13-digit values),
but the code was multiplying them by 1000, converting them to microseconds
and producing dates thousands of years in the future. This broke timestamp-based
sorting and display logic.

Verified with real API response showing submitTime and executeTime are already
in milliseconds format, matching firstObserved and lastUpdated fields.

jorbuedo committed:
Merge branch 'develop' into steelswap

jorbuedo committed:
fix: update steelswap transformers test to expect Portfolio token ID format

The transformer correctly converts Steelswap token IDs (no dot) to Portfolio format (with dot). Updated the test expectation to match the actual transformer behavior.

stackchain commented on mobile/packages/swap/adapters/api/steelswap/api-maker.ts:
```diff
@@ -0,0 +1,376 @@
+  const baseUrl = baseUrls[network]
+
+  // Cache for token decimals: Portfolio token ID (policyId.hexName format) -> decimals
+  const tokenDecimalsCache = new Map<Portfolio.Token.Id, number>()
```
really hope them can add to the API Q.Q

jorbuedo replied:
Working on it yeah

stackchain commented on mobile/packages/swap/adapters/api/steelswap/api-maker.ts:
```diff
@@ -0,0 +1,376 @@
+      tag,
+      error: {
+        ...error,
+        message: (() => {
```
doesnt need to be IIF (AI crap)

stackchain commented on mobile/packages/swap/adapters/api/steelswap/types.ts:
```diff
@@ -0,0 +1,194 @@
+
+export type TxStatus = 'txPending' | 'queued' | 'executed' | 'cancelled'
+
+export type OrderType = 0 | 1 | 2
```
what is 0 1 2? can we have a comment here?

jorbuedo replied:
Comes directly from their docs, doesn't really say what it means. I can ask, but transformer isn't really using it, it's just there for type completion.

stackchain reviewed (approved):
lgmt, is there anything that we need on UI when is limita and only steelswap is set? cc: @jorbuedo

jorbuedo commented:
> lgmt, is there anything that we need on UI when is limita and only steelswap is set? cc: @jorbuedo

@stackchain no, it's the same case for minswap as well, it'll just fail to find any pools and I believe show an error saying limit not supported

jorbuedo committed:
fix

jorbuedo committed:
feat(swap): gate adapters by partners config and implement Steelswap isFloat

- Gate adapter APIs to only work when partner code exists in partners config
- Make SwapSettings dynamically render switches from partners object keys
- Remove protocol/blockedProtocols from Minswap and Steelswap (no limit orders)
- Implement isFloat parameter for Steelswap API (always true)
- Remove decimals cache from Steelswap (handled by API with isFloat)
- Update YoroiConfig types to match actual config structure (all fields optional)
- Add all partner fields (dexhunter, muesliswap, minswap, steelswap) to config types
- Fix optional chaining in ActionsBanner for swap initialPair
- Bump swap package version to 7.2.0

jorbuedo committed:
fix: correct fee calculations and remove decimal conversions

- Minswap: Include aggregator_fee in totalFee calculation
- Steelswap: Remove decimal conversions (API now returns all values in decimal format when isFloat=true)
- Steelswap: Use top-level values for totals instead of nested splitGroup values
- UI: Display aggregator name in route (e.g., 'via Steelswap')

jorbuedo committed:
fix: prevent floating point precision issues when adding fees

Convert fees to lovelace (integers) before addition, then convert back to ADA to avoid precision loss

jorbuedo committed:
fix: correct steelswap transformers test expectations for volumeFee

- Update test mocks to use volumeFee: 1000 instead of 0.001
- Update test expectations to match transformer output (1000 instead of 0.001)
- Update transformer comments to reflect that volumeFee is passed through as-is when isFloat=true

jorbuedo committed:
Merge branch 'develop' into steelswap

jorbuedo committed:
bump types

jorbuedo committed:
Merge branch 'steelswap' of github.com:Emurgo/yoroi into steelswap

jorbuedo commented on mobile/packages/swap/manager.ts:
```diff
@@ -77,7 +96,7 @@ export const swapManagerMaker: Swap.ManagerMaker = ({
 
 const apiManagerMaker = (
-  adapters: Record<Swap.Aggregator, Swap.Api>,
+  adapters: Partial<Record<Swap.Aggregator, Swap.Api>>,
```
good point, no expectation to remove DH anytime soon though, I'll refactor that later

jorbuedo commented on mobile/src/features/Swap/useCases/SwapSettings/SwapSettings.tsx:
```diff
@@ -229,59 +243,47 @@ export const SwapSettings = () => {
-                    aggregator === 'auto'
-                      ? ['muesliswap', 'dexhunter', 'minswap']
-                      : 'auto',
+                    aggregator === 'auto' ? enabledAggregators : 'auto',
```
That's expected. Should have them, or be disabled.
